aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+
+Environment ``string``
+    Key: ``environment``. Defaults to "Production". May be set to any value. Predefined values include "Development" and "Staging" in addition to "Production". Values are not case sensitive. See :doc:`environments`. Set using the ``UseEnvironment`` method.
+
+.. code-block:: c#
+
+    .UseEnvironment("Development")
+
+.. note:: By default, the environment is read from the ``ASPNETCORE_ENVIRONMENT`` environment variable. When using Visual Studio, the environment will be based on settings in the *launchSettings.json* file.
+
+Server URLs ``string``
+    Key: ``urls``. Set to a semicolon (;) separated list of URL prefixes to which the server should response. For example, "http://localhost:123". The domain/host name can be replaced with "*" to indicate the server listen to any request coming to any IP address or host on the specified port (for example, "http://*:5000"). The protocol ("http://" or "https://") must be included with each URL.
+
+.. code-block:: c#
+
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+
+Startup Assembly ``string``
+    Key: ``startupAssembly``. Determines the assembly to search for the ``Startup`` class (unless specified using ``WebHostBuilder.UseStartup<StartupType>``, which overrides this setting). Set using non-generic ``UseStartup`` method.
+
+.. code-block:: c#
+
+    .UseStartup("StartupAssemblyName")
+
+Web Root ``string``
+    Key: ``webroot``. If not specified, Web Root will use the Content Root path. Determines default root location for static files to be served by the app (relative to Content Root). Set using ``UseWebRoot``.
+
+.. code-block:: c#
+
+    .UseWebRoot("public")
+
+A host can use :doc:`configuration` to set its values, which may be subsequently overridden. This is specified using the ``UseConfiguration`` method.
+
+.. code-block:: c#
+  :emphasize-lines: 6
+
+  public static void Main(string[] args)
+  {
+    var config = new ConfigurationBuilder()
+      .AddCommandLine(args)
+      .AddJsonFile("hosting.json", optional: true)
+      .Build();
+
+      var host = new WebHostBuilder()
+        .UseConfiguration(config)
+        .UseKestrel()
+        .Configure(app => 
+        {
+            app.Run(async (context) => await context.Response.WriteAsync("Hi!"));
+        })
+        .Build();
+
+    host.Run();
+  }
+        
+  var host = new WebHostBuilder()    
+      .UseConfiguration(config)
+      .Build();
+
+    
+Once a host is built using ``WebHostBuilder``, it is started by calling its ``Run`` extension method, which starts the web application and blocks the calling thread until the host shutdown.
+
+.. code-block:: c#
+
+    host.Run();
+
+You can run the host in a non-blocking manner by calling its ``Start`` method:
+
+.. code-block:: c#
+
+  using (host)
+  {
+    host.Start();
+    Console.ReadLine();
+  }
+
+There is also an extension method that will start the web host and listen on urls specified:
```

Comment:
Pass a list of URLs to the ``Start`` method and it will listen on URLs specified: